components/script/dom/bindings/codegen/CodegenRust.py

```diff
@@ -2968,22 +2968,35 @@ def __init__(self, descriptor, toJSONMethod):
     def definition_body(self):
         ret = ''
         interface = self.descriptor.interface
+
         for m in interface.members:
             if m.isAttr() and not m.isStatic() and m.type.isJSONType():
                 name = m.identifier.name
+                conditions = MemberCondition(None, None, m.exposureSet)
+                ret_conditions = 'vec![' + ",".join(conditions) + "]"
```

Comment:
Let's use `&[` instead of `vec![` to avoid allocating a bunch of memory needlessly. Also use `", "` when joining to make the resulting code a bit easier to read.

Reply:
thanks! fixed 74995a5287dc0af8afe62e9b537db5141b854765